frame/membership/src/migrations/v4.rs

```diff
@@ -90,23 +89,17 @@ pub fn pre_migrate<P: GetStorageVersion, N: AsRef<str>>(old_pallet_name: N, new_
 		.map_or(true, |next_key| next_key.starts_with(&old_pallet_prefix)));
 
 	let new_pallet_prefix = twox_128(new_pallet_name.as_bytes());
-	let storage_version_key =
-		[&new_pallet_prefix, &twox_128(STORAGE_VERSION_STORAGE_KEY_POSTFIX)[..]].concat();
-	// ensure nothing is stored in the new prefix.
-	assert!(
-		storage::next_key(&new_pallet_prefix).map_or(
-			// either nothing is there
-			true,
-			// or we ensure that it has no common prefix with twox_128(new),
-			// or isn't the storage version that is already stored using the pallet name
-			|next_key| {
-				!next_key.starts_with(&new_pallet_prefix) || next_key == storage_version_key
-			},
-		),
-		"unexpected next_key({}) = {:?}",
-		new_pallet_name,
-		HexDisplay::from(&storage::next_key(&new_pallet_prefix).unwrap()),
+	let storage_version_key = twox_128(STORAGE_VERSION_STORAGE_KEY_POSTFIX);
+
+	let mut new_pallet_prefix_iter = frame_support::storage::KeyPrefixIterator::new(
+		new_pallet_prefix.to_vec(),
+		new_pallet_prefix.to_vec(),
+		|key| Ok(key.to_vec()),
 	);
+
+	// Ensure nothing except the storage_version_key is stored in the new prefix.
+	assert!(new_pallet_prefix_iter.all(|key| key == storage_version_key));
```

Comment:
in the past we only ensured that the first key was sotrage_version_key, now we ensure it is the only one.

Reply:
tiny nit, but this doesn't check length, right? Would it be more accurate to do:

```
assert!(new_pallet_prefix_iter.next(), Some(storage_version_key));
assert!(new_pallet_prefix_iter.next(), None);
```
?

Reply:
yes it is true, but I was thinking if they change the name, maybe it is ok not to have anything stored at the new prefix.